src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolJudicialMappingTest.java

```diff
@@ -0,0 +1,270 @@
+        List<RoleAssignment> roleAssignments = new ArrayList<>();
+        QueryResults queryResults = (QueryResults) results.getValue(ROLE_ASSIGNMENTS_RESULTS_KEY);
+        for (QueryResultsRow row : queryResults) {
+            roleAssignments.add((RoleAssignment) row.get("$roleAssignment"));
+        }
+
+        //assertion
+        assertFalse(roleAssignments.isEmpty());
+        assertEquals(6, roleAssignments.size());
+        assertEquals("senior-judge",roleAssignments.get(0).getRoleName());
+        assertEquals("hmcts-judiciary",roleAssignments.get(1).getRoleName());
+        assertEquals("leadership-judge",roleAssignments.get(2).getRoleName());
+        assertEquals("case-allocator",roleAssignments.get(3).getRoleName());
+        assertEquals("task-supervisor",roleAssignments.get(4).getRoleName());
+        assertEquals("judge",roleAssignments.get(5).getRoleName());
+        assertEquals(judicialOfficeHolders.stream().iterator().next().getUserId(),roleAssignments.get(0).getActorId());
+        assertEquals(judicialOfficeHolders.stream().iterator().next().getUserId(),roleAssignments.get(1).getActorId());
+        assertEquals(judicialOfficeHolders.stream().iterator().next().getUserId(),roleAssignments.get(2).getActorId());
+        assertEquals(judicialOfficeHolders.stream().iterator().next().getUserId(),roleAssignments.get(3).getActorId());
+        assertEquals(judicialOfficeHolders.stream().iterator().next().getUserId(),roleAssignments.get(4).getActorId());
+        assertEquals(judicialOfficeHolders.stream().iterator().next().getUserId(),roleAssignments.get(5).getActorId());
+        assertEquals("SALARIED", roleAssignments.get(0).getAttributes().get("contractType").asText());
+        assertEquals("SALARIED", roleAssignments.get(1).getAttributes().get("contractType").asText());
+        assertEquals("SALARIED", roleAssignments.get(2).getAttributes().get("contractType").asText());
+        assertEquals("SALARIED", roleAssignments.get(3).getAttributes().get("contractType").asText());
+        assertEquals("SALARIED", roleAssignments.get(4).getAttributes().get("contractType").asText());
+        assertEquals("SALARIED", roleAssignments.get(5).getAttributes().get("contractType").asText());
+        assertEquals(workTypes, roleAssignments.get(0).getAttributes().get("workTypes").asText());
+        assertEquals(workTypesAccess, roleAssignments.get(2).getAttributes().get("workTypes").asText());
+        assertEquals(workTypes, roleAssignments.get(5).getAttributes().get("workTypes").asText());
+    }
+
+    @Test
+    void shouldReturnImmigrationJudgeRoles() {
+
+        judicialOfficeHolders.forEach(judicialOfficeHolder -> {
+            judicialOfficeHolder.setOffice("IAC Designated Immigration Judge");
+        });
+
+        //Execute Kie session
+        buildExecuteKieSession(getFeatureFlags("iac_1_1", true), false);
+
+        //Extract all created role assignments using the query defined in the rules.
```

Comment:
have you cover all scenarios based on excel sheet?

Reply:
can you send me this excel please?